pkg/dao/repository.go

```diff
@@ -84,29 +107,47 @@ func (r repositoryDaoImpl) fetchRepoConfig(orgID string, uuid string) (models.Re
 	return found, nil
 }
 
+func (r repositoryDaoImpl) fetchRepositoryWithURL(url string) (models.Repository, error) {
+	found := models.Repository{}
+	if err := r.db.Where("URL = ?", url).First(&found).Error; err != nil {
+		return found, err
+	}
+	return found, nil
+}
+
 func (r repositoryDaoImpl) Update(orgID string, uuid string, repoParams api.RepositoryRequest) error {
+	var repo models.Repository
+	if repoParams.URL != nil {
+		var err error
+		repo.URL = *repoParams.URL
+		if err = r.db.Where("URL = ?", repoParams.URL).FirstOrCreate(&repo).Error; err != nil {
+			return err
```

Comment:
Should probably wrap this in a call to DBErrorToApi

Reply:
Yes I have :)  thanks! updated!